# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/XPCOMInit.cpp
# Commit: 1f4dc2487e0e
# Full Hash: 1f4dc2487e0e822691d70c2b5337a6b664325752
# Author: Adam Brouwers-Harries <abrouwersharries@mozilla.com>
# Date: 2024-04-03 09:34:09
# Regressor Bug: 1865930
# File Overlap Count: 1
# Description:
#   Bug 1865930: Add support for stopping the profiler using UNIX signals r=canaltinova,profiler-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D203146
# ==============================================================================

diff -r 58a369871836 -r 1f4dc2487e0e xpcom/build/XPCOMInit.cpp
--- a/xpcom/build/XPCOMInit.cpp	Tue Apr 02 18:20:20 2024 +0000
+++ b/xpcom/build/XPCOMInit.cpp	Tue Apr 02 18:35:50 2024 +0000
@@ -325,6 +325,8 @@
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
+
+  // Initialise the profiler
   AUTO_PROFILER_INIT2;
 
   // Set up the timer globals/timer thread
@@ -464,6 +466,11 @@
   // to the directory service.
   nsDirectoryService::gService->RegisterCategoryProviders();
 
+  // Now that both the profiler and directory services have been started
+  // we can find the download directory, where the profiler can write
+  // profiles if necessary
+  profiler_lookup_download_directory();
+
   // Init mozilla::SharedThreadPool (which needs the service manager).
   mozilla::SharedThreadPool::InitStatics();
 
